libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+import * as pathUtilities from 'path';
+import * as fs from 'fs';
+import * as fsx from 'fs-extra';
+
+import { Text } from './';
+
+/**
+ * The allowed types of encodings, as supported by Node.js
+ * @public
+ */
+export type Encoding = 'utf8';
```

Comment:
>xport type Encoding = 'utf8'; [](start = 1, length = 29)

If people hardly ever specify a nondefault encoding, you could use a const enum:

```javascript
export const enum Encoding {
  Utf8 = 'utf8',
  Utf16 = 'utf16'
}

function f(encoding: Encoding): void {

}

f(Encoding.Utf16);
```